spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicComponentFactory.java

```diff
@@ -0,0 +1,198 @@
+	}
+
+	/**
+	 * Create the {@link DestinationTopicProcessor} that will be used to process the
+	 * {@link DestinationTopic} instances and store them in the provided
+	 * {@link DestinationTopicResolver}.
+	 * @param destinationTopicResolver the {@link DestinationTopicResolver}
+	 * instance to be used to store the {@link DestinationTopic} instances.
+	 * @return the instance.
+	 */
+	public DestinationTopicProcessor destinationTopicProcessor(DestinationTopicResolver destinationTopicResolver) {
+		return new DefaultDestinationTopicProcessor(destinationTopicResolver);
+	}
+
+	/**
+	 * Create the instance of {@link DestinationTopicResolver} that will be used to store
+	 * the {@link DestinationTopic} instance and resolve which a given record should be
+	 * forwarded to.
+	 * @return the instance.
+	 */
+	public DestinationTopicResolver destinationTopicResolver() {
+		return new DefaultDestinationTopicResolver(this.internalRetryTopicClock);
+	}
+
+	/**
+	 * Create the {@link DeadLetterPublishingRecovererFactory} that will be used to create
+	 * the {@link DeadLetterPublishingRecoverer} to forward the records to a given
+	 * {@link DestinationTopic}.
+	 * @param destinationTopicResolver the {@link DestinationTopicResolver} instance to
+	 * resolve the destinations.
+	 * @return the instance.
+	 */
+	public DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory(
+			DestinationTopicResolver destinationTopicResolver) {
+		return new DeadLetterPublishingRecovererFactory(destinationTopicResolver);
+	}
+
+	/**
+	 * Create the {@link ListenerContainerFactoryResolver} that will be used to resolve
+	 * the appropriate {@link KafkaListenerContainerFactory} for a given topic.
+	 * @param beanFactory the {@link BeanFactory} that will be used to retrieve the
+	 * {@link KafkaListenerContainerFactory} instance if necessary.
+	 * @return the instance.
+	 */
+	public ListenerContainerFactoryResolver listenerContainerFactoryResolver(BeanFactory beanFactory) {
+		return new ListenerContainerFactoryResolver(beanFactory);
+	}
+
+	/**
+	 * Create a {@link ListenerContainerFactoryConfigurer} that will be used to
+	 * configure the {@link KafkaListenerContainerFactory} resolved by the
+	 * {@link ListenerContainerFactoryResolver}.
+	 * @param kafkaConsumerBackoffManager the {@link KafkaConsumerBackoffManager} used
+	 * with the {@link KafkaBackoffAwareMessageListenerAdapter}.
+	 * @param deadLetterPublishingRecovererFactory the factory that will provide the
+	 * {@link DeadLetterPublishingRecoverer} instance to be used.
+	 * @param clock the {@link Clock} instance to be used with the listener adapter.
+	 * @return the instance.
+	 */
+	public ListenerContainerFactoryConfigurer listenerContainerFactoryConfigurer(KafkaConsumerBackoffManager kafkaConsumerBackoffManager,
+																				DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory,
+																				Clock clock) {
+
```

Comment:
Strange indentation.

Reply:
I think maybe GitHub counts tabs as 8 spaces instead of the 4 I'm used to. The goal was to align the additional parameters to the one in the method declaration - which is kind of how it looks in my IDE.

Either way, the pointer of using two tabs past `public` will save me a lot of moving things around 😄 Thanks.